### PR TITLE
Use Microsoft.Extensions.Http.Resilience

### LINKED ISSUE #14701: Replace Microsoft.Extensions.Http.Polly package with Microsoft.Extensions.Http.Resilience
Replace `Microsoft.Extensions.Http.Polly` package with [Microsoft.Extensions.Http.Resilience](https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.http.resilience?view=dotnet-plat-ext-8.0)

Microsoft.Extensions.Http.Resilience can be used with all .net versions we support (net6, net7 and net 8)

### PR DESCRIPTION
**Pull request by hishamco**
Fixes #14701

### CODE REVIEW

**Comment by hishamco** [comment]
@jtkech your review if you have a time

File: src/OrchardCore.Modules/OrchardCore.Twitter/Startup.cs [hunk 1]
```diff
@@ -47,7 +48,17 @@ public override void ConfigureServices(IServiceCollection services)
 
             services.AddHttpClient<TwitterClient>()
                 .AddHttpMessageHandler<TwitterClientMessageHandler>()
-                .AddTransientHttpErrorPolicy(policy => policy.WaitAndRetryAsync(3, attempt => TimeSpan.FromSeconds(0.5 * attempt)));
+                .AddResilienceHandler("oc-handler", builder => builder
+                    .AddRetry(new HttpRetryStrategyOptions
+                    {
+                        Name = "oc-retry",
```
Comment (MikeAlhayek) [inline src/OrchardCore.Modules/OrchardCore.Twitter/Startup.cs:54]: Maybe we should move the HttpRetryStrategyOptions into a static instance so we can reuse it in other places.

File: src/OrchardCore.Modules/OrchardCore.Twitter/Startup.cs [hunk 2]
```diff
@@ -47,7 +48,17 @@ public override void ConfigureServices(IServiceCollection services)
...
+                        MaxRetryAttempts = 3
+                    })
+                    .AddTimeout(new HttpTimeoutStrategyOptions
+                    {
+                        Name = "oc-timeout",
+                        Timeout = TimeSpan.FromSeconds(4),
```
Comment (MikeAlhayek) [inline src/OrchardCore.Modules/OrchardCore.Twitter/Startup.cs:60]: Same here. 4 seconds is different than the existing value. Not sure if this is okay or not. Probably is fine but not sure.

File: src/OrchardCore.Modules/OrchardCore.Twitter/Startup.cs (hunk 1 above)
Comment (hishamco) [reply src/OrchardCore.Modules/OrchardCore.Twitter/Startup.cs:54]: > so we can reuse it in other places

Which places do you mean?

**Review by MikeAlhayek — APPROVED** [review]

**Pull request merged**